lib/bootstrap/bootstrap.ts

```diff
@@ -28,6 +30,9 @@ export const bootstrap: FastifyPluginAsync<BootstrapConfig> = fp<BootstrapConfig
 
     if ('directory' in config) await autoLoadModules(config as AutoLoadConfig, fastify);
     if ('controllers' in config) loadControllers(config as ControllersListConfig, fastify);
+
+    // fixme the bootstrap should wait for this but it fails...
```

Comment:
What is the error?

Fastify uses [`avvio`](https://github.com/fastify/avvio) for plugins system and [here](https://github.com/fastify/fastify/blob/master/fastify.js#L292-L298) it's initialized in Fastify with [default = 10s](https://github.com/fastify/fastify/blob/master/lib/configValidator.js#L24)

Reply:
Something odd is happening there. If you await the `readyMap` the whole bootstrap times out.

The problem seems to be that the bootstrap function runs and is awaited before the Service instances are actually created (therefore entering a deadlock), but I can't seem to find where that order is settled in the code.

That await needs to happen exactly after all the Services are inited, but before the fastify server starts. Any ideas how/where I can do that?

Reply:
I'm trying to dig deeper into issue, my current theory this happens because library initialize controllers async but I'm giving up more and more with it 😕

Reply:
So I found root cause, it was because of async 😅

Reply:
I pushed commit with fix to your branch